tests/source/issue-6116/main.rs

```diff
@@ -0,0 +1,7 @@
+fn foo() -> fn(i32) -> i32 {
+    |a| {
+        ;
+        a
+    }
+}
+
```

Comment:
I'd like to expand on these test cases.

1. More than one statement in the block
    ```rust
    fn bar() -> fn(i32) -> i32 {
        |a| {
            ;
            a;
            b
        }
    }
    ```
2. Some tests where we have comments. For example:
   ```rust
   fn foo() -> fn(i32) -> i32 {
       |a| {
           /*comment before empty statement */;
           a
       }
   }
   ```
3. Any other edge cases that you can think of that we should consider.